### PR TITLE
bug: fix the IAM events not being tracked

### PR DESCRIPTION
**Pull request by leila-gu**
Fixed the IAM events not being tracked: 
- InAppMessaging_triggeredEvent : When event logging is done.
- InAppMessaging_impressions : When the campaign is shown to the user.

Updated the event name to following:
InAppMessaging_events  -> InAppMessaging_triggeredEvent

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data **before every commit**, including API endpoints and keys
- [x] I checked that all possible internal and platform/library exceptions are gracefully handled and **will not** crash the host app
- [x] I checked for open & known issues (especially crashes) for any newly added platform APIs or libraries
- [x] I tested non-trivial changes on a real device
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: analytics/src/main/java/com/rakuten/tech/mobile/analytics/Event.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ * An event to be tracked.
+ * @param name Event's name
+ * @param parameters Event parameters
+ */
+class Event(name: String, parameters: Map<String, Any>) {
+
+    /**
+     * Method to track an event.
+     */
+    fun track() {}
```
Comment (vincentdelmundo) [inline analytics/src/main/java/com/rakuten/tech/mobile/analytics/Event.kt:13]: Please fix the [code smells](https://sonarcloud.io/project/issues?id=rakutentech_android-inappmessaging&pullRequest=133&resolved=false&types=CODE_SMELL) here.

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/LegacyEventTrackerHelper.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+/**
+ * <p>Utility that allow to send events to analytics module for event tracking. If the Analytics module version
+ * 4.7.0 or newer is installed (i.e. included as compile or implementation dependency of the app)
+ * the event will be processed. If there is no compatible analytics module is present nothing
+ * happens. To request the Analytics module to track an event use {@link #sendEvent(String,
+ * Map)}. Events with names that have a `_` (underscore) as prefix will be processed by the RAT
+ * tracker & parameters will be sent as `cp` data (as of 4.7.0).
+ */
+internal object LegacyEventTrackerHelper {
```
Comment (vincentdelmundo) [inline inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/LegacyEventTrackerHelper.kt:15]: Please remove docs for this class.
Also, please remove "Legacy" in the class name.

**Review by vincentdelmundo — CHANGES_REQUESTED** [review]
Looks good. Just minor comments.

File: inappmessaging/src/test/java/com/rakuten/tech/mobile/analytics/Event.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.rakuten.tech.mobile.analytics
+
+/**
+ * An event to be tracked.
+ * @param name Event's name
+ * @param parameters Event parameters
+ */
+class Event(name: String, parameters: Map<String, Any>) {
```
Comment (imaginaris) [inline inappmessaging/src/test/java/com/rakuten/tech/mobile/analytics/Event.kt:8]: How is this class related to `Event` from `src/main` ? (That might be a newbie question)

File: inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/LegacyEventTrackerHelper.kt (hunk 2 above)
Comment (leila-gu) [reply inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/LegacyEventTrackerHelper.kt]: Done

File: inappmessaging/src/test/java/com/rakuten/tech/mobile/analytics/Event.kt (hunk 3 above)
Comment (leila-gu) [reply inappmessaging/src/test/java/com/rakuten/tech/mobile/analytics/Event.kt:8]: Hi @imaginaris,
Since the Event class from src/main is not recognized from the test resource file at compile-time, I added this fake Event class to generate it.

File: analytics/src/main/java/com/rakuten/tech/mobile/analytics/Event.kt (hunk 1 above)
Comment (leila-gu) [reply analytics/src/main/java/com/rakuten/tech/mobile/analytics/Event.kt]: Done

**Review by vincentdelmundo — APPROVED** [review]
LGTM

File: inappmessaging/src/test/java/com/rakuten/tech/mobile/analytics/Event.kt (hunk 3 above)
Comment (imaginaris) [reply inappmessaging/src/test/java/com/rakuten/tech/mobile/analytics/Event.kt:8]: Ok, got it. It doesn't look like a fake :D
